components/Rating.js

```diff
@@ -0,0 +1,49 @@
+const React = require('react');
+
+class Rating extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      increasing: false,
+      decreasing: false
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.rating < nextProps.rating) {
```

Comment:
This is a little verbose and has too much control flow, we can simplify this to:

```js
this.setState({
  increasing: this.props.rating < nextProps.rating,
  decreasing: this.props.rating > nextProps.rating
});
```